samples/smp_pi/README.rst

```diff
@@ -0,0 +1,40 @@
+.. _smp_pi:
+
+SMP Pi
+###########
+
+Overview
+********
+A simple application that calculates Pi independedntly in many threads.
+That allows to demonstrate a benefit of multiple execution units (CPU cores)
+when compute-intensive tasks could be run in parallel, i.e. there're
+no cross-dependencies or shared resources.
+
+With change of CONFIG_MP_NUM_CPUS on SMP systems one may see how almost
+linearly scales total execution time. With more cores in use less time it takes.
+
+Building and Running
+********************
+
+This project outputs Pi values calculated by each thread and in the end total time
+required for all the calculation to be done. It can be built and executed
+on QEMU as follows:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/smp_pi
+   :host-os: unix
+   :board: qemu_x86
+   :goals: run
+   :compact:
+
+Sample Output
+=============
+
+.. code-block:: console
+
+    Calculate first 10 digits of Pi independently by 4 thread(s).
```

Comment:
The sample code in this PR will output 240 digits by 16 threads (not 10 using 4 threads as shown here).

Reply:
@abrodkin is this sprintf issue resolved? It doesn't seem to be related to this test, did you create on issue for it?

Reply:
Well Actually I'm a bit lost here and at this point it looks like what I saw was just a memory corruption due to insufficient stack allocated to worker treads. Because after rebase on Zepyr master I started to see the same hang even without `sprintf()`. Then I added yet another 256 bytes to worker threads stack and now both implementations work.

But the problem is I cannot really debug that in x86_64 QEMU due to https://github.com/zephyrproject-rtos/sdk-ng/issues/123 so I may only speculate on that given for ARC it used to work in any way :(

Reply:
ok - stack corruption sounds as a plausible explanation. The fatal error prints don't give a clue on what's wrong ("unknown error"). Any stack guards/checks/canaries enabled in the test? Those should lead to more clear error messages on stack corruption.